### PR TITLE
CDAP-3560 change isSystem to a scope enum

### PR DESCRIPTION
**Pull request by albertshau**
Changing the isSystem boolean to an enum for artifact scope.
We will start out with system and user scopes, but this allows us
to add new scopes in the future and is consistent with other
concepts like metrics.

Also doing some cleanup so that ArtifactId is used in a few
classes instead of storing the individual id fields.

Also adding missing package-info file for artifacts.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT2707

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java [hunk 1]
```diff
@@ -135,13 +137,19 @@ public void refreshSystemArtifacts(HttpRequest request, HttpResponder responder)
...
+    throws NamespaceNotFoundException, BadRequestException {
 
     Id.Namespace namespace = validateAndGetNamespace(namespaceId);
 
     try {
-      responder.sendJson(HttpResponseStatus.OK, artifactRepository.getArtifacts(namespace, includeSystem));
+      if (scope == null) {
+        responder.sendJson(HttpResponseStatus.OK, artifactRepository.getArtifacts(namespace, true));
+      } else {
+        ArtifactScope artifactScope = validateScope(scope);
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java]: Can't we call validateAndGetNamespace to combine these two lines of code?

**Comment by gokulavasan** [comment]
One comment. Rest LGTM :+1:

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java (hunk 1 above)
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java]: good point will do

**Comment by gokulavasan** [comment]
LGTM :+1:

**Pull request merged**